### ISSUE TITLE
Integrate internal Java code to github

### ISSUE THREAD

**Issue by xfxyjwf** [protocolbuffers/protobuf#4566]
We have a new implementation of Java lite runtime that reduces generated code size substantially and heavily depends on the use of unsafe. The new implementation needs to be integrated into github before we can find a solution for the Java 9 unsafe issue.

**Comment by haon4** [comment]
Already done.
